### DISCUSSION TITLE
Please help config Proxychains-ng for Yi Home 17CN

### DISCUSSION THREAD

**Discussion by voquangminh** [alienatedsec/yi-hack-v5#273] [Q&A] [answered]
Hi,
I have Yi home 720p 17CN and success working with yi hack v3 by using proxychains-ng tool for a long time.

Now i had change a new router, and i can not config proxychains-ng anymore because old server are dying.

After i reset camera and intsall new yi hack v5 but cannot use it "This camera may be not use outside china"

I search so much web to config proxychains but can not find how to use China proxy
this is proxychains config file i use.

random_chain

chain_len = 1

tcp_read_time_out 15000
tcp_connect_time_out 8000

[ProxyList]
socks4 61.171.50.169 6688
socks4 218.78.65.202 6688

Please help me to use camera again. I had succeed connect it but always show offline and said check your internet connection.

**Comment by alienatedsec** [comment]
@voquangminh Have you tried changing this from `no` to `yes` in the `system.conf` file?

https://github.com/alienatedsec/yi-hack-v5/blob/4407928f3c1f65cabcca83e19c14d0d843983232/src/static/static/home/yi-hack-v5/etc/system.conf#L19

**Reply by voquangminh** [reply]
@alienatedsec yes, i had changed PROXYCHAINSNG=yes, and succeed setup camera into app but can not stream because network problem, i think the problem because proxy china is not working and i don't know how to check if proxychains-ng is working or not.
![download_proxy_list_completed_example](https://github.com/alienatedsec/yi-hack-v5/assets/18108859/2a9645a8-0e4e-443b-808b-66809a08e287)
as i remember yi hack v3 had the tool for checking this, can you help me with proxychains-ng config.

**Reply by alienatedsec** [reply]
I can try integrating the same in the next pre-release. Also, I will update the version if it exists.

**Reply by alienatedsec** [reply]
Please try the latest release `0.4.0a` but this may be limited to setting the config only

**Reply by voquangminh** [reply]
@alienatedsec thank u, I appreciated with that, I will try and update to you.

**Reply by voquangminh** [reply]
Hi, after update to pre-release ver and testing proxy but It is still not working, it does not show anything result after click save and test config.
![image](https://github.com/alienatedsec/yi-hack-v5/assets/18108859/2e63f52b-28e3-436b-a5ba-577b4818c32c)

**Reply by alienatedsec** [reply]
Please can you post the string config here?

**Reply by voquangminh** [reply]
Hi, this is string I use 
`socks4 61.171.50.169 6688;socks4 218.78.65.202 6688;socks5 1.180.0.242 7302;socks5 60.190.195.146 10800`

**Reply by alienatedsec** [reply]
It works - at least partially - so far it shows results for `Test without proxy:` 

I need to fix the proxychain execution as it feels like the final compilation is wrong - leave it with me, but you should see something after pressing `save and test configuration` button

**Reply by alienatedsec** [reply]
@voquangminh I fixed the execution bit, but it shows the following

```
/tmp/sd/yi-hack-v5/bin #  proxychains4 wget -O- -q http://ipinfo.io
[proxychains] config file found: /tmp/sd/yi-hack-v5/etc/proxychains.conf
[proxychains] preloading /tmp/sd/yi-hack-v5/lib/libproxychains4.so
[proxychains] DLL init: proxychains-ng 4.16
error: no valid proxy found in config
/tmp/sd/yi-hack-v5/bin # cat  /tmp/sd/yi-hack-v5/etc/proxychains.conf
socks4 61.171.50.169 6688
socks4 218.78.65.202 6688
socks5 1.180.0.242 7302
socks5 60.190.195.146 10800
/tmp/sd/yi-hack-v5/bin #
```

I will attach a separate package for you to test

**Reply by alienatedsec** [reply, marked as answer]
@voquangminh fixed - at least for me

![image](https://github.com/alienatedsec/yi-hack-v5/assets/73167064/bbd69743-0175-43c4-a405-26d5e484ca65)

Try and follow #278 or wait for `0.4.0b`
